upgrade_test.py

```diff
@@ -415,186 +415,205 @@ def test_rolling_upgrade(self):  # pylint: disable=too-many-locals,too-many-stat
-        complex_cs_thread_pool = self.run_stress_thread(
-            stress_cmd=stress_cmd_complex_verify_read, profile='data_dir/complex_schema.yaml')
-        # wait for the read complex workload to finish
-        self.verify_stress_thread(complex_cs_thread_pool)
-
-        # After adjusted the workloads, there is a entire write workload, and it uses a fixed duration for catching the data lose.
-        # But the execute time of workloads are not exact, so let only use basic prepare write & read verify for complex workloads,
-        # and comment two complex workloads.
-        #
-        # TODO: retest commented workloads and decide to enable or delete them.
-        #
-        # complex workload: verify data by multiple ops
-        #self.log.info('Starting c-s complex workload to verify data by multiple ops')
-        #stress_cmd_complex_verify_more = self.params.get('stress_cmd_complex_verify_more')
-        #complex_cs_thread_pool = self.run_stress_thread(stress_cmd=stress_cmd_complex_verify_more, profile='data_dir/complex_schema.yaml')
-
-        # wait for the complex workload to finish
-        # self.verify_stress_thread(complex_cs_thread_pool)
-
-        # complex workload: verify data by delete 1/10 data
-        #self.log.info('Starting c-s complex workload to verify data by delete')
-        #stress_cmd_complex_verify_delete = self.params.get('stress_cmd_complex_verify_delete')
-        #complex_cs_thread_pool = self.run_stress_thread(stress_cmd=stress_cmd_complex_verify_delete, profile='data_dir/complex_schema.yaml')
-
-        # wait for the complex workload to finish
-        # self.verify_stress_thread(complex_cs_thread_pool)
-
-        error_factor = 3
-        schema_load_error_num = 0
-
-        for node in self.db_cluster.nodes:
-            errors = node.search_database_log(search_pattern='Failed to load schema version',
-                                              start_from_beginning=True,
-                                              publish_events=False)
-            schema_load_error_num += len(errors)
-
-        self.log.debug('schema_load_error_num: %d', schema_load_error_num)
-        assert schema_load_error_num <= error_factor * 8 * \
-            len(self.db_cluster.nodes), 'Only allowing shards_num * %d schema load errors per host during the entire test, actual: %d' % (
-                error_factor, schema_load_error_num)
-
-        self.log.debug('start sstabledump verify')
-        self.db_cluster.nodes[0].remoter.run('for i in `sudo find /var/lib/scylla/data/keyspace_complex/ -type f |grep -v manifest.json |grep -v snapshots'
-                                             ' |head -n 1`; do echo $i; sudo sstabledump $i 1>/tmp/sstabledump.output || exit 1; done', verbose=True)
-
-        self.log.info('all nodes were upgraded, and last workaround is verified.')
+        with self.subTest('pre-test - prepare test kesyapces and tables'):
```

Comment:
subTest only exists for unittest.TestCase of Python3.
We can't use it for upgrade of 2019.1

Reply:
good reason to move to python3 :)